content/en/blog/_posts/2023-08-04-kubernetes-1.28-blog.md

```diff
@@ -0,0 +1,306 @@
+
+- PostStart and PreStop lifecycle handlers for sidecar containers
+- All probes (startup, readiness, liveness)
+    
+Readiness probes of sidecars will contribute to determining the whole Pod readiness.
+    
+## Beta support for enabling swap space on Linux
+    
+This adds swap support to nodes in a controlled, predictable manner so that Kubernetes users can perform testing and provide data to continue building cluster capabilities on top of swap.
+
+There are two distinct types of users for swap, who may overlap:
+
+- Node administrators, who may want swap available for node-level performance tuning and stability/reducing noisy neighbor issues.
+    
+- Application developers, who have written applications that would benefit from using swap memory. There are hence a number of possible ways that one could envision swap use on a node.
+   
+## Add Support To Handle Non-graceful Node Shutdown
+    
+This feature allows stateful workloads to restart on a different node if the original node is shut down unexpectedly or ends up in a non-recoverable state perhaps due to hardware failure or unresponsive OS.
+
+This allows stateful workloads to failover to a different node successfully after the original node is shut down or in a non-recoverable state such as the hardware failure or broken OS.
+    
+The Graceful Node Shutdown introduces a way to detect a node shutdown and handle it gracefully. However, a node shutdown action may not be detected by Kubelet's Node Shutdown Manager, either because the command does not trigger the inhibitor locks mechanism used by Kubelet or because of a user error, i.e., the ShutdownGracePeriod and ShutdownGracePeriodCriticalPods are not configured properly.
+
+When a node is shutdown but not detected by Kubelet's Node Shutdown Manager, the pods that are part of a StatefulSet will be stuck in terminating status on the shutdown node and cannot move to a new running node. This is because Kubelet on the shutdown node is not available to delete the pods so the StatefulSet cannot create a new pod with the same name. If there are volumes used by the pods, the VolumeAttachments will not be deleted from the original shutdown node so the volumes used by these pods cannot be attached to a new running node. As a result, the application running on the StatefulSet cannot function properly. If the original shutdown node comes up, the pods will be deleted by Kubelet and new pods will be created on a different running node. If the original shutdown node does not come up, these pods will be stuck in terminating status on the shutdown node forever.
+
+This handles node shutdown cases that are not detected by Kubelet's Node Shutdown Manager. The pods will be forcefully deleted in this case, triggering the deletion of the VolumeAttachments, and new pods will be created on a new running node so that application can continue to function.
+
+## Automatic, retroactive assignment of a default StorageClass
```

Comment:
The section here is too long really. Can we summarize or link to a previous release announcement?